src/main/java/io/confluent/connect/elasticsearch/jest/JestElasticsearchClient.java

```diff
@@ -187,13 +190,16 @@ public Version getVersion() {
   }
 
   private boolean indexExists(String index) {
-    Action action = new IndicesExists.Builder(index).build();
-    try {
-      JestResult result = client.execute(action);
-      return result.isSucceeded();
-    } catch (IOException e) {
-      throw new ConnectException(e);
+    if (!indexCache.keySet().contains(index)) {
```

Comment:
couple of points here:

1. Since indexCache is only tested for the presence of a value, we probably only need a HashSet<> here. 
2. Making a keySet everytime is expensive. We can check for membership with `containsKey(key)`.

Reply:
fair point, thanks @wicknicks here are my rusty java skills at display.